daemon/prune.go

```diff
@@ -70,11 +72,20 @@ func (daemon *Daemon) VolumesPrune(config *types.VolumesPruneConfig) (*types.Vol
 }
 
 // ImagesPrune removes unused images
-func (daemon *Daemon) ImagesPrune(config *types.ImagesPruneConfig) (*types.ImagesPruneReport, error) {
+func (daemon *Daemon) ImagesPrune(pruneFilters filters.Args) (*types.ImagesPruneReport, error) {
 	rep := &types.ImagesPruneReport{}
 
+	danglingOnly := true
+	if pruneFilters.Include("dangling") {
+		if pruneFilters.ExactMatch("dangling", "false") || pruneFilters.ExactMatch("dangling", "0") {
+			danglingOnly = false
+		} else if !(pruneFilters.ExactMatch("dangling", "true") || pruneFilters.ExactMatch("dangling", "1")) {
+			return nil, errors.New("Flag --filter dangling= should only be true or false")
```

Comment:
Says it must be true or false here, but above we accept `0` for false?
Is there precedent in other filters for this?
Most boolean params we accept true/false/1/0